File: android_fragment/commons-core/src/main/java/com/common/core/utils/SharedPreferencesUtils.java
package com.common.core.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.Nullable;

import java.util.Map;
import java.util.Set;

public class SharedPreferencesUtils {
    private static final String NAME_SHAREPREFERENCES = "share_preferences";
    private SharedPreferences mSharedPreferences;
    private SharedPreferences.Editor mEditor;

    private SharedPreferencesUtils(Context context, String name) {
        mSharedPreferences = context.getSharedPreferences(name, Context.MODE_PRIVATE);
        mEditor = mSharedPreferences.edit();
    }

    public static SharedPreferencesUtils get(Context context, String name) {
        return new SharedPreferencesUtils(context, name);
    }

    public static SharedPreferencesUtils get(Context context) {
        return new SharedPreferencesUtils(context, NAME_SHAREPREFERENCES);
    }

    public Map<String, ?> getAll() {
        return mSharedPreferences.getAll();
    }

    @Nullable
    public String getString(String s, String s1) {
        return mSharedPreferences.getString(s, s1);
    }

    @Nullable
    public Set<String> getStringSet(String s, Set<String> set) {
        return mSharedPreferences.getStringSet(s, set);
    }

    public int getInt(String s, int i) {
        return mSharedPreferences.getInt(s, i);
    }

    public long getLong(String s, long l) {
        return mSharedPreferences.getLong(s, l);
    }

    public float getFloat(String s, float v) {
        return mSharedPreferences.getFloat(s, v);
    }

    public boolean getBoolean(String s, boolean b) {
        return mSharedPreferences.getBoolean(s, b);
    }

    public boolean contains(String s) {
        return mSharedPreferences.contains(s);
    }

    public void putString(String key, String value) {
        mEditor.putString(key, value);
        mEditor.commit();
    }

    public void putStringSet(String key, Set<String> values) {
        mEditor.putStringSet(key, values);
        mEditor.commit();
    }

    public void putInt(String key, int value) {
        mEditor.putInt(key, value);
        mEditor.commit();
    }

    public void putLong(String key, long value) {
        mEditor.putLong(key, value);
        mEditor.commit();
    }

    public void putFloat(String key, float value) {
        mEditor.putFloat(key, value);
        mEditor.commit();
    }

    public void putBoolean(String key, boolean value) {
        mEditor.putBoolean(key, value);
        mEditor.commit();
    }

    public void remove(String key) {
        mEditor.remove(key);
        mEditor.commit();
    }

    public void clear() {
        mEditor.clear();
        mEditor.commit();
    }
}

File: example/commons-core/src/main/java/com/common/core/utils/SharedPreferencesUtils.kt
package com.common.core.utils

import android.content.Context
import android.content.SharedPreferences

class SharedPreferencesUtils private constructor(context: Context, name: String) {
    private val mSharedPreferences: SharedPreferences
    private val mEditor: SharedPreferences.Editor

    val all: Map<String, *>
        get() = mSharedPreferences.all

    init {
        mSharedPreferences = context.getSharedPreferences(name, Context.MODE_PRIVATE)
        mEditor = mSharedPreferences.edit()
    }

    fun getString(s: String, s1: String): String? {
        return mSharedPreferences.getString(s, s1)
    }

    fun getStringSet(s: String, set: Set<String>): Set<String>? {
        return mSharedPreferences.getStringSet(s, set)
    }

    fun getInt(s: String, i: Int): Int {
        return mSharedPreferences.getInt(s, i)
    }

    fun getLong(s: String, l: Long): Long {
        return mSharedPreferences.getLong(s, l)
    }

    fun getFloat(s: String, v: Float): Float {
        return mSharedPreferences.getFloat(s, v)
    }

    fun getBoolean(s: String, b: Boolean): Boolean {
        return mSharedPreferences.getBoolean(s, b)
    }

    operator fun contains(s: String): Boolean {
        return mSharedPreferences.contains(s)
    }

    fun putString(key: String, value: String) {
        mEditor.putString(key, value)
        mEditor.commit()
    }

    fun putStringSet(key: String, values: Set<String>) {
        mEditor.putStringSet(key, values)
        mEditor.commit()
    }

    fun putInt(key: String, value: Int) {
        mEditor.putInt(key, value)
        mEditor.commit()
    }

    fun putLong(key: String, value: Long) {
        mEditor.putLong(key, value)
        mEditor.commit()
    }

    fun putFloat(key: String, value: Float) {
        mEditor.putFloat(key, value)
        mEditor.commit()
    }

    fun putBoolean(key: String, value: Boolean) {
        mEditor.putBoolean(key, value)
        mEditor.commit()
    }

    fun remove(key: String) {
        mEditor.remove(key)
        mEditor.commit()
    }

    fun clear() {
        mEditor.clear()
        mEditor.commit()
    }

    companion object {
        private val NAME_SHAREPREFERENCES = "share_preferences"

        operator fun get(context: Context, name: String): SharedPreferencesUtils {
            return SharedPreferencesUtils(context, name)
        }

        operator fun get(context: Context): SharedPreferencesUtils {
            return SharedPreferencesUtils(context, NAME_SHAREPREFERENCES)
        }
    }
}